tests/php/Forms/FormTest.php

```diff
@@ -646,16 +670,27 @@ public function testDisableSecurityTokenAcceptsSubmissionWithoutToken()
             count($tokenEls ?? []),
             'Token form field added for controller without disableSecurityToken()'
         );
-        $token = (string)$tokenEls[0];
-        $response = $this->submitForm(
+
+        [ $form, $button, $data, $withSecurityToken ] = [ ...$formData ];
+        
+        $response = $this->submitForm($form, $button, $data, $withSecurityToken);
+        $this->assertEquals($statusCode, $response->getStatusCode(), $testMessage);
```

Comment:
Sorry, I should have been clearer in https://github.com/silverstripe/silverstripe-framework/pull/10643#discussion_r1097963999 - I think this whole section (everything that relies on the dataprovider) should be its own separate method (i.e. declare a new method for the stuff that uses the dataprovider, including the missing button example)

```php
public function testWhateverYouWantToCallIt($form, $button, $data, $withSecurityToken, $expectedStatusCode, $testMessage)
{
        if (is_null($button)) {
            $response = $this->submitForm($form, $button, $data, $withSecurityToken);
            $this->assertEquals($expectedStatusCode, $response->getStatusCode(), $testMessage);
        } else {
            // Test nonexistent button Exceptions
            $this->expectException(\Exception::class);
            $this->expectExceptionMessage($testMessage);

            $this->submitForm($form, $button, $data, $withSecurityToken);
        }
}
```